velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -722,6 +747,12 @@ TypePtr ReaderBase::convertType(
 
   static std::string_view kTypeMappingErrorFmtStr =
       "Converted type {} is not allowed for requested type {}";
+  // If 'requestedType' is or contains an array type, skip the check of element
```

Comment:
How does this catch a valid column mismatch case? We should still check if the Parquet Schema has 2-level  or 3-level array schema encoding.